### PR TITLE
[SYCL] Configure ARM cross-compilation

### PR DESCRIPTION
**Pull request by Rbiessy**
Need more tests on AMD first.

### CODE REVIEW

File: third_party/sycl/crosstool/CROSSTOOL.tpl [hunk 1]
```diff
@@ -35,65 +40,172 @@ toolchain {
   tool_path { name: "compat-ld" path: "/usr/bin/ld" }
   tool_path { name: "cpp" path: "/usr/bin/cpp" }
   tool_path { name: "dwp" path: "/usr/bin/dwp" }
-  tool_path { name: "gcc" path: "computecpp" }
-  # Use "-std=c++11" for nvcc. For consistency, force both the host compiler
-  # and the device compiler to use "-std=c++11".
-  cxx_flag: "-std=c++11"
-  linker_flag: "-Wl,-no-as-needed"
-  linker_flag: "-lstdc++"
-  linker_flag: "-B/usr/bin/"
```
Comment (lukeiwanski) [inline third_party/sycl/crosstool/CROSSTOOL.tpl:44]: That's needed.

**Comment by lukeiwanski** [comment]
Overall great work @Rbiessy  && @DuncanMcBain !
We might want to add `-no-serial-memop` as the compiler is noisy about it.

File: third_party/sycl/crosstool/CROSSTOOL.tpl (hunk 1 above)
Comment (Rbiessy) [reply third_party/sycl/crosstool/CROSSTOOL.tpl:44]: Stuff moved but everything should be there now!

**Comment by Rbiessy** [comment]
Thanks! Yes I am adding it on the command line. I'm just not sure if we always want this option? I don't mind adding it if you want.

**Comment by lukeiwanski** [comment]
Adding it via command line makes sense!

**Comment by DuncanMcBain** [comment]
I can't remember the platforms that you need to avoid it on. You might be able to suppress the remarks with something like -Rno-serial-memop without actually generating the different code.

**Pull request merged**